### PR TITLE
Add background settings for paragraphs group component

### PR DESCRIPTION
**Pull request by darimpulso**
Add Basic_BodyWidhtBackground Component

    - Choose a background color (default: white / grey / black)
    - Upload a background image
    - Add margin to bottom
    - extandable with Kwc_Parallax

### CODE REVIEW

File: trl.xml [hunk 1]
```diff
@@ -13519,5 +13519,60 @@ Odprite aktivacijsko povezavo v brskalniku. Potem vnesite geslo, ki ga boste v p
...
+<id>2261</id>
+</text>
+<text>
+<en>Parallax velocity</en>
+<de>Parallax Geschwindigkeit</de>
+<id>2262</id>
+</text>
+<text>
+<en>Parallax velocity lower than 1 = faster than scrolling (ex: 0,5 = 200%)</en>
+<de>Parallax Geschwindigkeit kleiner than 1 = schneller als scrolling (zb: 0,5 = 200%)</de>
```
Comment (yacon) [inline trl.xml]: "kleiner than" ;-)

File: Kwc/Paragraphs/Group/BackgroundSettings/ParallaxEffect.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+(function() {
```
Comment (yacon) [inline Kwc/Paragraphs/Group/BackgroundSettings/ParallaxEffect.js]: onLoad?

File: Kwc/Basic/BackgroundWindowWidth/Component.scss [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        &.backgroundImage {
+            z-index: 1;
+            .image {
+                position: relative;
+                width: 100%;
+                height: 100%;
+                background-position: center -50%;
+            }
+            &.parallax {
```
Comment (nsams) [inline Kwc/Basic/BackgroundWindowWidth/Component.scss]: remove parallax related styling

File: Kwc/Basic/BackgroundWindowWidth/Component.twig [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<div class="{{ cssClass }} {{ backgroundColor }}">
+    {% if image.hasContent() %}
+        {{ renderer.component(image) }}
+    {% endif %}
+    <div class="background {{ backgroundColor }}{% if image.hasContent() %} opaque{% endif %}"></div>
```
Comment (nsams) [inline Kwc/Basic/BackgroundWindowWidth/Component.twig]: why output backgroundColor twice?

File: Kwc/Basic/BackgroundWindowWidth/Form.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<?php
+class Kwc_Basic_BackgroundWindowWidth_Form extends Kwc_Abstract_Composite_Form
+{
+    protected function _initFields()
+    {
+        parent::_initFields();
+
+        $types = array(
```
Comment (nsams) [inline Kwc/Basic/BackgroundWindowWidth/Form.php]: make types configurable thru component setttings

File: Kwc/Basic/BackgroundWindowWidth/Form.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            'none' => 'None',
+            'white' => trlKwf('White'),
+            'grey' => trlKwf('Grey'),
+            'black' => trlKwf('Black')
+        );
+        $fs = $this->add(new Kwf_Form_Container_FieldSet(trlKwf('Standard Adjustments')));
+        $fs->add(new Kwf_Form_Field_Select('background_color', trlKwf('Background color')))
+            ->setAllowBlank(true)
+            ->setDefaultValue(array_shift(array_keys($types)))
```
Comment (nsams) [inline Kwc/Basic/BackgroundWindowWidth/Form.php]: that doesn't work, remove it

File: Kwc/Basic/BackgroundWindowWidth/Form.php [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+            ->setValues($types);
+        $fs->add(new Kwf_Form_Field_NumberField('margin_bottom', trlKwf('Margin bottom')))
+            ->setAllowBlank(true)
+            ->setDefaultValue(0);
```
Comment (nsams) [inline Kwc/Basic/BackgroundWindowWidth/Form.php]: that doesn't work, remove it

File: trl.xml [hunk 8]
```diff
@@ -13519,5 +13519,65 @@ Odprite aktivacijsko povezavo v brskalniku. Potem vnesite geslo, ki ga boste v p
...
+<de>Parallax Effekt</de>
+<id>2261</id>
+</text>
+<text>
+<en>Parallax speed</en>
+<de>Parallax Geschwindigkeit</de>
+<id>2262</id>
+</text>
+<text>
+<en>Parallax velocity lower than 1 = faster than scrolling (ex: 0,5 = 200%)</en>
```
Comment (nsams) [inline trl.xml]: remove pralallax stuff from trl

**Comment by nsams** [comment]
+1

**Pull request merged**